src/Angor/Client/Pages/Invest.razor

```diff
@@ -1234,6 +1254,29 @@ else
         return false;
     }
 
+    private async Task SetMaxInvestment()
+    {
+        var maxAmount = WalletBalanceBtc ;
+        if (maxAmount < 0) maxAmount = 0;
+
+        Investment.InvestmentAmountBtc = Math.Round(maxAmount, 8, MidpointRounding.ToZero);
```

Comment:
I think just set the flag that says max, then I will dop the rest in the method that is building the trx